druid/src/widget/either.rs

```diff
@@ -55,7 +56,11 @@ impl<T: Data> Widget<T> for Either<T> {
         }
     }
 
-    fn update(&mut self, ctx: &mut UpdateCtx, _old_data: Option<&T>, data: &T, env: &Env) {
+    fn lifecycle(&mut self, ctx: &mut LifeCycleCtx, event: &LifeCycle, data: &T, env: &Env) {
```

Comment:
A discussion question: should this widget be generating lifecycle events when the children are hidden and shown in response to a data change? In any case, it makes sense to preserve the current behavior for now, and enhance this later. (We do want to get this sorted out in preparation for really getting the tab widget right)

The non-shown widget will not be getting lifecycle events, so there's a risk of invariants not being maintained.

Reply:
yea, my expectation is that we will use (new?) lifecycle events for that case.